llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -2121,6 +2120,36 @@ bool AArch64InstructionSelector::preISelLower(MachineInstr &I) {
     }
     return false;
   }
+  case TargetOpcode::G_INSERT_VECTOR_ELT: {
+    // The tablegen patterns for integer insert_vector_elts with small elements
+    // will expect the operand to be legalized to a i32 type. This extends the
+    // types of i8 and i16 element inserts, so that the existing tablegen
+    // patterns can apply.
```

Comment:
This is doing legalization work. The selector code should be as simple as possible (i.e., you shouldn't have to preprocess anything for the patterns to work)

Reply:
I think this needs to be after reg-bank selection. Otherwise there isn't a good way to detect INSERT_VECTOR_ELT from gpr values vs fpr registers for the same type size.
The process is irtranslation -> opt -> legalization -> opt -> regbank select -> regbank-dependant legalization -> selection. And currently that lives in this preISelLower method. I'm not sure if it's the best place for it, but there are other operations like select that could do with post-reg-back-select legalization to treat grp selects differently to fpr.

Reply:
regbank-dependant legalization is part of regbankselect; the apply mapping should apply any regbank related legalization work

Reply:
If it can avoid this manual selection of the ANYEXT then it sounds good. There didn't seem to be anything already like it in AArch64, but I've added something for it. Thanks.